### PR TITLE
Add register.sh script and call from install.py

### PR DESCRIPTION
**Pull request by danjoneslf**
+ update readme to explain registration actions.

### CODE REVIEW

**Review by gds-ahine — APPROVED** [review]
👍

**Comment by 0atman** [comment]
```shell
./install.py 

✔️ Detect-secrets hook installed
Enter your github username:
0atman
Performing GitHub OAuth
This creates a personal access token with read:user and read:org scopes. 
Please enter your GitHub 2FA code:
338004
Requesting authorization from GitHub - You will be prompted for your GitHub password.
Enter host password for user '0atman':
Submit registration request.
You have been registered successfully.
✔️ User registered
```
:+1:

**Review by 0atman — APPROVED** [review]

File: register.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+# Register and get reporting credentials
+echo "Submit registration request."
+response=$(curl -s -H "Authorization: github ${token}" -H "User-Agent: GitHub/Hook" -d '{"action":"register"}' https://${endpoint}?alert_name=register)
+
+# Check the response is JSON
+response_type=$(echo $response | jq -r type)
+if [[ $response_type == "object" ]]; then
+  secret=$(echo $response | jq -r '.user_secret')
+  username=$(echo $response | jq -r '.username')
+
```
Comment (alice-carr) [inline register.sh:71]: Mine failed first time round as I don't have jq installed - and we can't guarantee that everyone will have jq installed / or want to install it

Comment (0atman) [reply register.sh:71]: +1 for this, it's a shame to use jq, as the rest of this repo only relies on `python` and `curl`, I think.

Comment (0atman) [reply register.sh:71]: I have great news, we can do without it! 

```
cat my.json | python -m json.tool
```
Python has a built-in tool for pretty-printing json, which will exit with a non-zero code if it's invaild! :ok_hand:

Comment (0atman) [reply register.sh:71]: oh, but you can't slice off the secret and username with that... python rewrite?

File: install.py [hunk 2]
```diff
@@ -34,3 +34,13 @@ def _subprocess_call(cmd):  # this is the python 2.7 implementation
 )
 print()
 print("✔️ Detect-secrets hook installed")
+
+_subprocess_call(
+    [
+        "bash",
+        os.environ["HOME"] + "/.gds-pre-commit/register.sh",
```
Comment (0atman) [inline install.py]: this no longer is how this file is used, @akinnane and @OllieJC made the repo relocatable

**Review by 0atman — DISMISSED** [review]

File: register.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+import json
+import sys
+
+from runner import run
+
+try:
+    from past.builtins import raw_input  # type: ignore
+except ImportError:
+    pass
```
Comment (0atman) [inline register.py:12]: @danjoneslf I've fixed it in the way I think is best. This'll fail python 2 syntax check, but - critically - pass our dev flake8!

**Pull request merged**